diffkemp/semdiff/caching.py

```diff
@@ -55,13 +55,17 @@ class Vertex:
         Note: names, files and lines are tuples containing the values for both
         modules.
         """
-        def __init__(self, names, result, files=None, lines=None):
+        def __init__(self, names, result, files=None, lines=None,
+                     inst_cnt=None, inst_equal_cnt=None, lines_cnt=None):
             self.names = names
             self.result = result
             self.nonfun_diffs = []
             self.successors = ([], [])
             self.files = files
             self.lines = lines
+            self.inst_cnt = inst_cnt
+            self.inst_equal_cnt = inst_equal_cnt
+            self.lines_cnt = lines_cnt
```

Comment:
I'm wondering if it's worth enclosing these into a separate object like `stats`. It could save some lines, especially if we add more stats in future, and also better represents the purpose of these fields.

The encapsulation would probably have to be done in SimpLL, too, so it's a bit of extra work.

Reply:
Good idea, I will change this. I am not really sure about saving some lines. The `Stats` object/class in the frontend could load the sub-yaml dynamically but I think it will still be a bit more maintainable to explicitly state the `Stats` attributes in the frontend.

Reply:
Done